parquet-avro/src/main/java/org/apache/parquet/avro/AvroRecordConverter.java

```diff
@@ -59,6 +62,10 @@
  */
 class AvroRecordConverter<T> extends AvroConverters.AvroGroupConverter {
 
+  private static final String STRINGABLE_PROP = "avro.java.string";
+  private static final String JAVA_CLASS_PROP = "java-class";
+  private static final String JAVA_KEY_CLASS_PROP = "java-key-class";
```

Comment:
Those are from Avro. I was avoiding having them in here, but I needed them to mimic Avro's String deserialization. Most of the time, we use a call to Avro's `getClass(Schema)` rather than using them directly, but the method we need isn't public.